HUB-18210 Setup database to allow client certificate authentication

romeara opened the pull request:
Enable features such as foreign data wrappers between databases within
the container by generating and storing client certificates for
authentication

Commit:
HUB-18210 Setup database to allow client certificate authentication

Enable features such as foreign data wrappers between databases within
the container by generating and storing client certificates for
authentication

romeara commented:
Note: I do not have authorization to merge this. Please either grant permission or merge once approved

paykin commented on hub-database.sh:
```diff
@@ -87,12 +102,87 @@ else
+  	
+  	# Convert client key to usable format
+    mv /tmp/blackduck_user.crt /tmp/hub-db-user.crt
+    rm -f /tmp/blackduck_user.crt
```
Why `rm` if you do `mv`?

romeara replied:
To keep the file state consistent - I don't want to leave errant files around. Unless... wait, mv is effectively rename, isn't it (facepalm)

romeara replied:
Extra "rm"'s removed

paykin commented:
What is "data wrappers between databases within the container"?

romeara commented:
@paykin Foreign data wrappers - allows setup of one database to read data from another. Specifically, we are going to use this to make a much more efficient reporting database process

Commit:
Remove extraneous "rm" commands

There were multiple remove commands following move commands that had the same result of a rename

blackduck-cookt commented on hub-database.sh:
```diff
@@ -31,8 +31,9 @@ echo "Database host: $targetDatabaseHost"
 ## trigger this each time when starting a container.
 ## (needs to be kept in sync with 2-hub-setup.sh)
-if [ -d "$PGDATA" ] && [ -s "$PGDATA/PG_VERSION" ];
+if [ -d "$PGDATA" ] && [ -s "$PGDATA/$PG_VERSION" ];
```
This change is not correct.  PG_VERSION is a file that is created in PGDATA by initdb.  This test is checking that the data directory exists and that initdb has been run.

romeara replied:
Change reverted, should be using "PG_VERSION" in all instances now

blackduck-cookt commented on hub-database.sh:
```diff
@@ -48,14 +49,26 @@ hostnossl all             	all                   0.0.0.0/0                 rejec
-## Certificate management
+dataPopulated=false
+
+if [ -d "$PGDATA" ] && [ -s "$PGDATA/$PG_VERSION" ];
```
PG_VERSION, not $PG_VERSION

romeara replied:
Corrected

blackduck-cookt commented on hub-database.sh:
```diff
@@ -48,14 +49,26 @@ hostnossl all             	all                   0.0.0.0/0                 rejec
+
+if [ -d "$PGDATA" ] && [ -s "$PGDATA/$PG_VERSION" ];
+then
+	echo "Data directory not populated"
```
I think the "not" is in the wrong branch.

romeara replied:
Corrected

Commit:
Fix $PG_VERSION back to PG_VERSION

Commit:
Fix data directory state logging

bd-skatzman commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM blackducksoftware/hub-docker-common:1.0.1 as docker-common
+FROM blackducksoftware/hub-docker-common:1.0.3 as docker-common
```
I am assuming a subsequent PR on hub-backend will bump the versions?

romeara replied:
Of this container, yes
